### PR TITLE
win32: pump messages during manager_thread join to prevent STA deadlock

### LINKED ISSUE #521: [Windows] Shutdown deadlock: manager_thread.join() blocks STA message pump
### Operating System Info
Windows 11

### Other OS
_No response_

### OBS Studio Version
Other

### OBS Studio Version (Other)
32.1.2

### OBS Studio Log URL
_N/A — not a log-visible issue; OBS simply hangs indefinitely_

### OBS Studio Crash Log URL
_N/A — no crash, just a hang requiring force-kill_

### Expected Behavior
OBS exits cleanly within a few seconds of closing the main window.

### Current Behavior
OBS hangs indefinitely on exit when third-party plugins have created Qt widgets that register COM shell proxies (e.g., any plugin opening a `QDialog` or `QDockWidget`). The hang is in `obs_module_unload` → `manager_thread.join()` → `WaitForSingleObject(INFINITE)`.

The root cause: on Windows, the main thread is an STA COM host. During `CefShutdown()`, CEF's `CrBrowserMain` thread calls `CoUninitialize`, which attempts cross-apartment `Release` calls that require the main thread to pump messages. Since `manager_thread.join()` blocks without pumping, this deadlocks.

macOS is unaffected because it uses the `ENABLE_BROWSER_QT_LOOP` path, which runs `BrowserShutdown()` inline on the main thread.

### Steps to Reproduce
1. Install any OBS plugin that opens a Qt dialog (e.g., obs-ptz, StreamElements, etc.)
2. Launch OBS
3. Open and interact with the plugin's dialog/dock
4. Close OBS via File → Exit or window close button
5. OBS hangs ~80% of the time (must be force-killed via Task Manager)

### Anything else we should know?

**Debugger confirmation** — the deadlock stack:
```
Main thread:    obs_module_unload → Thrd_join → WaitForSingleObject(hThread, INFINITE)
Manager thread: CefRunMessageLoop → BrowserShutdown → CefShutdown
  └→ CrBrowserMain: CoUninitialize → cross-apartment Release → SendMessage(main) → BLOCKED
```

**Fix validated with IAT hook proof-of-concept:**
A test plugin that patches `WaitForSingleObjectEx` in obs-browser.dll's IAT to use `MsgWaitForMultipleObjects` + message pump loop eliminates the hang entirely:
- Before: 4/5 shutdowns HUNG (60s timeout, force-killed)
- After: 14/14 shutdowns PASS (~1s clean exit)

**Suggested fix** (in `obs-browser-plugin.cpp`):
Replace `manager_thread.join()` on Windows with:
```cpp
#ifdef _WIN32
    HANDLE hThread = (HANDLE)manager_thread.native_handle();
    while (true) {
        DWORD res = MsgWaitForMultipleObjects(1, &hThread, FALSE, INFINITE, QS_ALLINPUT);
        if (res == WAIT_OBJECT_0)
            break;
        if (res == WAIT_OBJECT_0 + 1) {
            MSG msg;
            while (PeekMessage(&msg, NULL, 0, 0, PM_REMOVE)) {
                TranslateMessage(&msg);
                DispatchMessage(&msg);
            }
        } else {
            break;
        }
    }
    manager_thread.detach();
#else
    manager_thread.join();
#endif
```

This allows COM dispatch calls to complete while waiting for the manager thread to exit.

### PR DESCRIPTION
**Pull request by sammallon**
Fixes #521

On Windows, `obs_module_unload` calls `manager_thread.join()` which blocks the main thread without pumping messages. During `CefShutdown()`, CEF's `CrBrowserMain` thread needs cross-apartment COM dispatch via the main thread's STA message pump. This deadlocks.

**Fix:** Replace `join()` with `MsgWaitForMultipleObjects` + message pump loop on Windows. After the thread signals, `detach()` prevents `std::thread`'s destructor from double-waiting.

**Testing:** Validated with an IAT hook PoC plugin that applies this same change at runtime:
- Before: 4/5 shutdowns hung indefinitely (60s timeout, force-killed)
- After: 14/14 shutdowns exited cleanly (~1s)

Tested on OBS 32.1.2, Windows 11.

### CODE REVIEW

**Comment by Warchamp7** [comment]
Please fill out the PR template.

**Comment by Fenrirthviti** [comment]
We do not accept AI generated submissions. Please ensure you take the time to read our project's contribution policy.
